### PR TITLE
Improve react-tagsinput definitions

### PR DESCRIPTION
**Pull request by mykter**
Based on https://github.com/samwise-tech/components (MIT licence)

Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] Run `npm run lint package-name` (or `tsc` if no `tslint.json` is present).

Select one of these and delete the others:

If changing an existing definition:
- [x] Provide a URL to documentation or source code which provides context for the suggested changes: No changes to the component, just improvements to the definitions - https://github.com/samwise-tech/components/blob/master/src/own-typings/react-tagsinput/index.d.ts
- [N/A] Increase the version number in the header if appropriate.
- [N/A] If you are making substantial changes, consider adding a `tslint.json` containing `{ "extends": "dtslint/dt.json" }`.

### CODE REVIEW

File: types/react-tagsinput/index.d.ts [hunk 1]
```diff
@@ -9,18 +9,42 @@ import * as React from "react";
 export as namespace ReactTagsInput;
 export = TagsInput;
 
+type Tag = any;
```
Comment (aozgaa) [inline types/react-tagsinput/index.d.ts]: Can it really be anything? Or is it `string | object` or just `string` in practice? Can you add a test with a real example?

**Review by paulvanbrenk — CHANGES_REQUESTED** [review]
I agree with @aozgaa I'm not convinced `Tag` should be of type `any`

**Comment by mykter** [comment]
you're right, it should be `string|object`.
However when I do this and add the corresponding example, I get these errors with the tags[].

The example where the tags are strings errors with:
```
    Types of property 'onChange' are incompatible.
      Type '(tags: string[]) => void' is not assignable to type '(tags: Tag[], changed: Tag[], changedIndexes: number[]) => void'.
        Types of parameters 'tags' and 'tags' are incompatible.
          Type 'Tag[]' is not assignable to type 'string[]'.
            Type 'Tag' is not assignable to type 'string'.
              Type 'object' is not assignable to type 'string'.
```
and the example where the tags are objects has the same error but the other type:
```
            Type 'Tag' is not assignable to type 'AnObject'.
              Type 'string' is not assignable to type 'AnObject'.
```

Any suggestions?

It looks like it's trying to assign the `tags: Array<Tag>` type from the onChange signature to the example implementation, rather than the other way round. So a string is assignable to `string|object` but not the other way round.

**Comment by paulvanbrenk** [comment]
@mykter going to leave that question to @aozgaa

**Pull request merged**